inference-engine/src/mkldnn_plugin/nodes/mkldnn_depth_to_space_node.cpp

```diff
@@ -141,37 +122,53 @@ void MKLDNNDepthToSpaceNode::initSupportedPrimitiveDescriptors() {
     }
     supportedTypes.push_back(LayoutType::ncsp);
     auto creators = BlockedDescCreator::getCommonCreators();
-    auto range = BlockedDescCreator::makeFilteredRange(creators, nDims, supportedTypes);
+    auto range = BlockedDescCreator::makeFilteredRange(creators, inputDataShape.getRank(), supportedTypes);
 
     for (auto itr = range.first; itr != range.second; ++itr) {
-        config.inConfs[0].desc = itr->second->createSharedDesc(precision, getInputShapeAtPort(0));
-        config.outConfs[0].desc = itr->second->createSharedDesc(precision, getOutputShapeAtPort(0));
+        config.inConfs[0].desc = itr->second->createSharedDesc(precision, inputDataShape);
+        config.outConfs[0].desc = itr->second->createSharedDesc(precision, outputDataShape);
         supportedPrimitiveDescriptors.emplace_back(config, impl_type);
     }
 }
 
 void MKLDNNDepthToSpaceNode::createPrimitive() {
-    auto &dstMemPtr = getChildEdgeAt(0)->getMemoryPtr();
-    auto &srcMemPtr = getParentEdgeAt(0)->getMemoryPtr();
+    dstMemPtr = getChildEdgeAt(0)->getMemoryPtr();
+    srcMemPtr = getParentEdgeAt(0)->getMemoryPtr();
     if (!dstMemPtr || !dstMemPtr->GetPrimitivePtr())
         THROW_ERROR << "has not allocated destination memory";
     if (!srcMemPtr || !srcMemPtr->GetPrimitivePtr())
         THROW_ERROR << "has not allocated input memory";
     if (getSelectedPrimitiveDescriptor() == nullptr)
         THROW_ERROR << "has unidentified preferable primitive descriptor";
+    attrs.dataSize = getSelectedPrimitiveDescriptor()->getConfig().inConfs[0].desc->getPrecision().size();
 
-    VectorDims srcDims = srcMemPtr->getStaticDims();
+    if (inputShapesDefined()) {
+        if (needPrepareParams())
+            prepareParams();
+        updateLastInputDims();
+    }
+}
 
+void MKLDNNDepthToSpaceNode::prepareParams() {
+    execPtr = std::make_shared<DepthToSpaceExecutor>(attrs, getParentEdgeAt(0)->getMemoryPtr(), getChildEdgeAt(0)->getMemoryPtr());
+}
+
+MKLDNNDepthToSpaceNode::DepthToSpaceExecutor::DepthToSpaceExecutor(const DepthToSpaceAttrs& attrs,
+                                                                   const MKLDNNMemoryPtr& srcMemPtr,
+                                                                   const MKLDNNMemoryPtr& dstMemPtr) {
```

Comment:
From my point of view passing both Memory or MemoryDesc as an executor parameter is a bad idea. You only uses small part of that structure to initialize the executor. My proposal is to include minimal required subset into DepthToSpaceAttrs. Then DepthToSpaceAttrs will contain enough data for key computation during caching.
@maxnick what do you think?

Reply:
As far, as I can see, we do not need to pass the memory objects. Moreover, `VectorDims dstDims = dstMemPtr->getStaticDims();` is not used at all. So it seems that all we need is `srcDims`, `srcBlockedDims` (I'm wondering whether just  `srcBlockedDims` could be enough), and the `LayoutType`.